Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1654,6 +1654,33 @@ std::unique_ptr<RenderStyle> RenderElement::getUncachedPseudoStyle(const Style::
     return WTFMove(resolvedStyle->style);
 }
 
+std::unique_ptr<RenderStyle> RenderElement::textSegmentPseudoStyle(PseudoId pseudo) const
+{
+    if (isAnonymous())
+        return nullptr;
+
+    if (auto pseudoStyle = getUncachedPseudoStyle({ pseudo })) {
```

Comment:
Why not use getCachedPseudoStyle? The reason we can't for highlight is because of the parameter name ::highlight(...)

Reply:
I was following what's already done for `::selection`.

We have the following comment above the methods:
```
    // The pseudo element style can be cached or uncached.  Use the cached method if the pseudo element doesn't respect
    // any pseudo classes (and therefore has no concept of changing state).
```

Though from testing, I don't think that `::selection` works with other pseudo classes either? Maybe @anttijk can weigh in on what the right thing to do here is.

Reply:
I don't know what that comment means. getCachedPseudoStyle is the right choice if it works. getUncachedPseudoStyle should only be needed if you have to pass additional parameters (like `::highlight()` name).